core/network-libp2p/src/service_task.rs

```diff
@@ -575,15 +577,20 @@ impl Service {
 				KadConnectionType::CannotConnect => continue,
 			};
 
-			self.topology.add_kademlia_discovered_addrs(
+			let changed = self.topology.add_kademlia_discovered_addrs(
```

Comment:
```suggestion
			topology_has_changed = self.topology.add_kademlia_discovered_addrs(
```

Reply:
We call `add_kademlia_discovered` multiple times, and we want `topology_has_changed` to be true if any of them returned `true`. This would only grab the latest.

Reply:
Ahh sorry, then `topology_has_changed |=` should do the trick :)

Reply:
Sorry that will also not work. Brain not working! Sorry!